### PR TITLE
Add layers for Gen4

### PR DESCRIPTION
**Pull request by rshym**
Components are copied from prod-devel-rcar-gen4.

### CODE REVIEW

File: meta-xt-gateway/meta-xt-domd-gen4/recipes-connectivity/snort/snort/configure.in-disable-tirpc-checking-for-fedora-snort-20.patch
Comment (lorc) [inline meta-xt-gateway/meta-xt-domd-gen4/recipes-connectivity/snort/snort/configure.in-disable-tirpc-checking-for-fedora-snort-20.patch:1]: I believe, this is a product-specific patch. We don't need snort support in the common parts.

Also, there is nothing rcar-specific in those changes

File: meta-xt-gateway/meta-xt-domd-gen4/recipes-connectivity/snort/snort/snort-tsn0.service
Comment (lorc) [inline meta-xt-gateway/meta-xt-domd-gen4/recipes-connectivity/snort/snort/snort-tsn0.service:1]: Those changes should go to prod-devel for sure.

File: meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend [hunk 1]
```diff
@@ -0,0 +1,14 @@
+IMAGE_INSTALL:append = " \
+    pciutils \
+    devmem2 \
+    iccom-support \
+    optee-test \
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend:5]: This should go to prod-devel

File: meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend [hunk 2]
```diff
@@ -0,0 +1,14 @@
+IMAGE_INSTALL:append = " \
+    pciutils \
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend:2]: This should go to prod-devel

File: meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    devmem2 \
+    iccom-support \
+    optee-test \
+"
+
+IMAGE_INSTALL:append = " iproute2 iproute2-tc tcpdump nvme-cli"
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend:8]: Most of those changes should go to prod-devel

File: meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+IMAGE_INSTALL:append = " kernel-module-nvme-core kernel-module-nvme"
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend:10]: Those as well

File: meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+IMAGE_INSTALL:append = " kernel-module-ixgbe"
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend:12]: And those

File: meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+IMAGE_INSTALL:append = " e2fsprogs"
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domd-gen4/recipes-core/images/rcar-image-minimal.bbappend:14]: Those too

File: meta-xt-gateway/meta-xt-domu-gen4/recipes-connectivity/vmq-network/files/tsn1.network [hunk 7]
```diff
@@ -0,0 +1,5 @@
+[Match]
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domu-gen4/recipes-connectivity/vmq-network/files/tsn1.network:1]: This is prod-devel thing

File: meta-xt-gateway/meta-xt-domu-gen4/recipes-connectivity/vmq-network/files/vmq0.network [hunk 8]
```diff
@@ -0,0 +1,6 @@
+[Match]
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domu-gen4/recipes-connectivity/vmq-network/files/vmq0.network:1]: This too

File: meta-xt-gateway/meta-xt-domu-gen4/recipes-connectivity/vmq-network/vmq-network.bb [hunk 9]
```diff
@@ -0,0 +1,23 @@
+SUMMARY = "Internal virtual network based on Renesas R-Switch VMQ feature"
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domu-gen4/recipes-connectivity/vmq-network/vmq-network.bb:1]: This whole recipe is meaningful only for prod-devel

File: meta-xt-gateway/meta-xt-domu-gen4/recipes-core/images/rcar-image-minimal.bbappend [hunk 10]
```diff
@@ -0,0 +1,15 @@
+IMAGE_INSTALL:append = " \
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domu-gen4/recipes-core/images/rcar-image-minimal.bbappend:1]: This whole file is prod-devel specific

File: meta-xt-gateway/meta-xt-domu-gen4/recipes-kernel/linux/linux-renesas/dmatest.cfg [hunk 11]
```diff
@@ -0,0 +1 @@
+CONFIG_DMATEST=m
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domu-gen4/recipes-kernel/linux/linux-renesas/dmatest.cfg:1]: I am not sure that we need this at all. And for sure not in the common code.

File: meta-xt-gateway/meta-xt-domu-gen4/recipes-kernel/linux/linux-renesas/ixgbevf.cfg [hunk 12]
```diff
@@ -0,0 +1 @@
+CONFIG_IXGBEVF=m
```
Comment (lorc) [inline meta-xt-gateway/meta-xt-domu-gen4/recipes-kernel/linux/linux-renesas/ixgbevf.cfg:1]: To prod-devel off we go

File: meta-xt-gateway/meta-xt-domu-gen4/recipes-kernel/linux/linux-renesas/r8a779f0-spider-domu.dts
Comment (lorc) [inline meta-xt-gateway/meta-xt-domu-gen4/recipes-kernel/linux/linux-renesas/r8a779f0-spider-domu.dts:1]: This whole DTS is very prod-devel-centric. I don't think we need this in the common layer

**Comment by rshym** [comment]
I have squashed three original commits into one "Introduce layer meta-xt-gateway for Gen4".
All fixes that I made to address comments are placed into two fixups, that are intended to be squashed into the first commit.

What was done:
- remove snort, daq, vmq;
- remove rcar-image-minimal, so product decides what to
  include into the image;
- remove domu, as it is not needed for now;
- remove xen, as we use quite different recipes now
  for 4.17 release and pre-4.17;
- remove optee-os;
- partially rework optee-test;
- move components from -domx-gen4 directory to -domd-gen4;
- domd's linux-renesas is stripped of product-specific things like DTS and some patches.

Tested with the corresponding PR for meta-aos-rcar-gen4.

**Review by lorc — APPROVED** [review]
Thanks
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by rshym** [comment]
Ack is added.
Fixups are squashed.
Ready to be merged.

**Pull request merged**